### ISSUE TITLE
Service Entry displayed in a different namespace in the graph

### ISSUE THREAD

**Issue by xeviknal** [kiali/kiali#3798]
I am following this istio task: [Egress gateway for https traffic](https://istio.io/latest/docs/tasks/traffic-management/egress/egress-gateway/#egress-gateway-for-https-traffic)

I realized that the Service Entry is created in the `default` namespace but it is displayed in the `istio-system` within the graph.

![Screenshot of Kiali (14)](https://user-images.githubusercontent.com/613814/112139072-9c809e80-8bd2-11eb-96fe-dfb3fcedd4ed.png)

![Screenshot of Kiali (15)](https://user-images.githubusercontent.com/613814/112139086-a2767f80-8bd2-11eb-8a55-e43dff7c1b26.png)

```
$ k get serviceentries.networking.istio.io --all-namespaces
NAMESPACE   NAME   HOSTS                 LOCATION   RESOLUTION   AGE
default     cnn    ["edition.cnn.com"]              DNS          73m
```

The bug is more evident when doing right-click into the service entry and choosing the details option:
![Screenshot of Kiali (16)](https://user-images.githubusercontent.com/613814/112139371-faad8180-8bd2-11eb-8a5d-b5667f811ab0.png)

![Screenshot of Kiali (17)](https://user-images.githubusercontent.com/613814/112139384-fd0fdb80-8bd2-11eb-90f7-681d1838e320.png)

Is it part of https://github.com/kiali/kiali/issues/3765?

**Comment by jshaughn** [comment]
@xeviknal I thought (by default) the service entry was exported to all namespaces, and as such it would be locally available in each.

**Comment by xeviknal** [comment]
@jshaughn yes they are. Perhaps I am missing some progress in here.
It was a bit weird that I couldn't see the SE when showing only the default namespace where it belongs but appears when showing the istio-system.

![Screenshot of Kiali (18)](https://user-images.githubusercontent.com/613814/112162450-5afced00-8bec-11eb-86a4-ed7109d34c9b.png)

![Screenshot of Kiali (19)](https://user-images.githubusercontent.com/613814/112162463-5df7dd80-8bec-11eb-8b45-4545a67461fd.png)

I missed that while browsing only `default` the egress node says it leaves in another namespace. I expected to see a node from external namespace send traffic to a service entry leaving in the browsing namespace.

I also expected that the link from the SE node to the details page would work too.

**Comment by jshaughn** [comment]
> Is it part of #3765?

@xeviknal , I'm not sure about some of the things reported here but I do think there is a bug in the context menu navigation, as you show above.  The issue is that we are looking for the service entry config in the "exportedTo" namespace, not the definition namespace.  I'll look at fixing that problem, I'm not sure about the other questions, I'd need to see the telemetry.

**Comment by jshaughn** [comment]
@nrfox , I started working in this and generated a [server-side PR](https://github.com/kiali/kiali/pull/3980).  I was about to start the UI side but then thought maybe it would be a good way for you to start getting familar with the front-end code.  So I co-assigned you.  I think the server-side thing I did is what you'd need but feel free to change things.  Also, we can meet if you like to do a walk-through of some of the relevant graph/ui code.

We are trying to solve the issue I mentioned in the comment above, not necessarily everything Xavi has entered into this issue.

**Comment by nrfox** [comment]
@jshaughn perfect! I was just looking for something.

**Comment by nrfox** [comment]
The bug in the context menu navigation has been fixed in #3980 and kiali/kiali-ui#2146. @jshaughn is there any other work to be done for this issue? It seems like the rest of what's covered is working as intended for how Service Entries are exported and displayed in the namespaces they are exported to.
